### ISSUE TITLE
git module hangs in vagrant shared folder

### ISSUE THREAD

**Issue by sp3c73r2038** [ansible/ansible#6789]
##### Issue Type:

"Bug Report"
##### Ansible Version:

ansible 1.6 (devel 57acf104c6) last updated 2014/04/01 10:46:13 (GMT +800)

devel branch, fetched directly from this repo.
##### Environment:
- host: (Gentoo) Linux
- vagrant guest: ubuntu precise 12.04 x86
- vagrant version: 1.4.3
##### Summary:

git module hangs when dest target is set to vagrant shared folder like `/vagrant/something`
##### Steps To Reproduce:

I have this in tasks

```
- name: fetch carabiner source code
  git: repo=git://git.corp.anjuke.com/_aps/carabiner
       dest={{ carabiner_home }}
       version=master
       force={{ carabiner_source_force_update }}
  sudo: yes
  sudo_user: {{ aps_username }}
```

and for vagant, `carabiner_home` is set to `/vagrant/.src/carabiner`, `carabiner_source_force_update` is set to `no`.

run `vagrant provision`
##### Expected Results:

source cloned / updated normally.
##### Actual Results:

ansible-playbook just hangs at that git step. debug output info:

```
TASK: [carabiner | fetch carabiner source code] ******************************* 
<192.168.123.2> ESTABLISH CONNECTION FOR USER: vagrant
<192.168.123.2> REMOTE_MODULE git repo=git://git.corp.anjuke.com/_aps/carabiner dest=/vagrant/.src/carabiner version=master force=False
<192.168.123.2> EXEC ['ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/konoha/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'Port=22', '-o', 'IdentityFile=/home/konoha/.vagrant.d/insecure_private_key', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=vagrant', '-o', 'ConnectTimeout=10', '192.168.123.2', "/bin/sh -c 'mkdir -p /tmp/ansible-tmp-1396338834.27-27071851497981 && chmod a+rx /tmp/ansible-tmp-1396338834.27-27071851497981 && echo /tmp/ansible-tmp-1396338834.27-27071851497981'"]
<192.168.123.2> PUT /tmp/tmpgMvg9k TO /tmp/ansible-tmp-1396338834.27-27071851497981/git
<192.168.123.2> EXEC ['ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/konoha/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'Port=22', '-o', 'IdentityFile=/home/konoha/.vagrant.d/insecure_private_key', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=vagrant', '-o', 'ConnectTimeout=10', '192.168.123.2', "/bin/sh -c 'chmod a+r /tmp/ansible-tmp-1396338834.27-27071851497981/git'"]
<192.168.123.2> EXEC ['ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/konoha/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'Port=22', '-o', 'IdentityFile=/home/konoha/.vagrant.d/insecure_private_key', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=vagrant', '-o', 'ConnectTimeout=10', '192.168.123.2', "/bin/sh -c '/usr/bin/python /tmp/ansible-tmp-1396338834.27-27071851497981/git'"]
```

The url is git protocol, so I assume there is nothing to do with `ssh-agent` thing.

Anything I could provide more to help locate the problem ?

TIA.

**Comment by mpdehaan** [comment]
You may want to consider accept_hostkey=yes as it's probably waiting on a hostkey.   Can you let us know if that helps solve the problem?

If you can reproduce this outside of vagrant with accept_hostkey=yes, that would be useful info.

**Comment by jctanner** [comment]
@aleiphoenix if the accept_hostkey parameter does not work, it would also be helpful to see where in the module code it is hanging. This mailing list post will walk you through that process:

https://groups.google.com/forum/#!msg/ansible-project/JdZWzLeZMmk/93W-42MP9U0J

**Comment by sp3c73r2038** [comment]
All right, ansible is just ok. 

The issue is after the repo is cloned, I've changed remote url to ssh protocol.

Thanks @jctanner , I run ansible-playbook with `ANSIBLE_KEEP_REMOTE_FILES=1` and get the git module file. trace leads to 

```
subprocess.py(1364):             while fd2file:
subprocess.py(1365):                 try:
subprocess.py(1366):                     ready = poller.poll()
The authenticity of host 'git.corp.anjuke.com (192.168.1.61)' can't be established.
ECDSA key fingerprint is 3b:7d:66:1b:11:64:4b:cb:c9:54:8d:15:92:01:1e:5f.
Are you sure you want to continue connecting (yes/no)? 
```

It hangs because there's no ssh key in guest system.

**Comment by jctanner** [comment]
@aleiphoenix are you checkout out the repo and changing the url within your playbook? accept_hostkey would solve the "Are you sure you want to ..." error.

**Comment by sp3c73r2038** [comment]
@jctanner No, I changed the remote url by hand, not in playbook. I almost forget that. :(

In my case , `known_hosts` is just fine :) `accept_hostkey=yes` works as expected. 

But as far as I've got no ssh key on guest system, fetching source code won't work (with the ssh url). So It's not ansible but my fault.

Thanks for the debugging tip.

I guess this issue is resolved and should be closed. Thanks.
